main/FatesInterfaceMod.F90

```diff
@@ -476,6 +478,12 @@ subroutine allocate_bcin(bc_in, nlevsoil_in, nlevdecomp_in, num_lu_harvest_cats)
 
       allocate(bc_in%pft_areafrac(maxpft))
```

Comment:
After testing the code myself, I noticed that this better to be `allocate(bc_in%pft_areafrac(**0:**maxpft))`. Then, it will be consistent with the indexing in the code in "https://github.com/rosiealice/ctsm/blob/SP_api/src/utils/clmfates_interfaceMod.F90" Line 630-639. The mismatch of the indexing can cause segmentation fault without showing where the problem happens (when DEGUB=false).

Reply:
Good catch @huitang-earth.  This is due to the hlm side code where the index for `pft_areafrac` is set to always index from zero regardless of the setting of `natpft_lb`: https://github.com/glemieux/ctsm/blob/6323556f1d1a4d4b9b40be993b814f7db2574de7/src/utils/clmfates_interfaceMod.F90#L630-L632

I also noticed that the `hlm_sp_` prefixed variables have the same indexing, so I changed those allocations to match as well: https://github.com/glemieux/fates/blob/3ac64584b5442791a16b119b265d827bad35d11f/main/FatesInterfaceMod.F90#L482-L486

Reply:
Actually I think I'm going to reverse course on this: I just double-checked and noticed that in EDInitMod and EDPhysiologyMod that the `pft_areafrac` and `hlm_pft_map` are using the same indexing variable which starts at index one to conform to the `hlm_pft_map` allocation: https://github.com/rosiealice/fates/blob/c7e4df8e01fe66a2dacd89527849c7c1d9a4629b/main/EDInitMod.F90#L324-L327